### PR TITLE
fix: incomplete call leaves a phantom note, no note on inferred function

### LINKED ISSUE #394: inferred functions change note messages
```py
def f(a=1): pass
f(x=1)  # error

def g(a: int = 1): pass # note: "g" defined here
g(x=1)  # error
```

### LINKED ISSUE #395: incomplete call can cause phantom note to appear
```py
def g(a=1): pass  # type: ignore[no-untyped-def]

g(
)# type: ignore[no-untyped-call]
```
```
👉 mypy test.py
test.py:3:1: note: Type is "def (a: int =) -> Untyped"
Success: no issues found in 1 source file
```

### PR DESCRIPTION
**Pull request by KotlinIsland**
Please include an amongus reference in this PR.

- resolves #394
- resolves #395

### CODE REVIEW

**Comment by KotlinIsland** [comment]
@DetachHead This is perfect for review

File: test-data/unit/check-based-incomplete-defs.test [hunk 1]
```diff
@@ -31,3 +31,16 @@ allow_incomplete_defs = True
...
+# flags: --python-version 3.8
+# This test is to check that the `notes` param on `mypy.errors.Errors.report` works correctly
+def f(a, b: int = 1): ...  # type: ignore[no-untyped-def]
+f(
+1)  # type: ignore[no-untyped-call]
+[out]
+main:4: error: Call to incomplete function "f" in typed context  [no-untyped-call]
+main:4: note: Type is "def (a: Untyped, b: int =) -> None"
+main:5: error: Unused "type: ignore" comment  [unused-ignore]
+[out version>=3.8]
```
Comment (DetachHead) [inline test-data/unit/check-based-incomplete-defs.test:46]: what about scenarios where the `type:ignore` works even tho its on the next line

Comment (KotlinIsland) [reply test-data/unit/check-based-incomplete-defs.test:46]: That is this scenario.

Comment (DetachHead) [reply test-data/unit/check-based-incomplete-defs.test:46]: why the `unused-ignore` error then?

Comment (KotlinIsland) [reply test-data/unit/check-based-incomplete-defs.test:46]: 3.7 doesn't support ignores on the wrong line, so it's expecting errors. But the `[out version>=3.8]` check is expecting no outputs.

**Review by DetachHead — APPROVED** [review]

**Pull request merged**